cpp/src/groupby/sort/group_std.cu

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "group_reductions.hpp"
+
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <cudf/column/column_factories.hpp>
+
+#include <thrust/reduce.h>
+#include <thrust/iterator/discard_iterator.h>
+
+namespace cudf {
+namespace experimental {
+namespace groupby {
+namespace detail {
+
+namespace {
+
+struct var_functor {
+  template <typename T>
+  std::enable_if_t<std::is_arithmetic<T>::value, std::unique_ptr<column> >
+  operator()(
+    column_view const& values,
+    column_view const& group_means,
+    rmm::device_vector<size_type> const& group_labels,
+    rmm::device_vector<size_type> const& group_sizes,
+    size_type ddof,
+    rmm::mr::device_memory_resource* mr,
+    cudaStream_t stream)
+  {
+    // TODO (dm): Use target_type and replace all reference to `double`
+    const size_type* d_group_labels = group_labels.data().get();
+    const size_type* d_group_sizes = group_sizes.data().get();
+    auto values_view = column_device_view::create(values);
+    auto means_view = column_device_view::create(group_means);
+    std::unique_ptr<column> result =
+      make_numeric_column(data_type(type_id::FLOAT64), group_sizes.size(),
+        mask_state::UNINITIALIZED, stream, mr);
+
+    auto values_it = thrust::make_transform_iterator(
+      thrust::make_counting_iterator(0),
+      [d_values=*values_view, d_means=*means_view, d_group_labels, d_group_sizes, ddof]
+      __device__ (size_type i) {
+        if (d_values.is_null(i))
+          return 0.0;
+        
+        double x = d_values.element<T>(i);
+        size_type group_idx = d_group_labels[i];
+        size_type group_size = d_group_sizes[group_idx];
+        
+        // prevent divide by zero error
+        if (group_size == 0 or group_size - ddof <= 0)
+          return 0.0;
+
+        double mean = d_means.element<double>(group_idx);
+        return (x - mean) * (x - mean) / (group_size - ddof);
+      }
```

Comment:
The value operated upon is not the `int` from the `transform_iterator`, it is taken from `d_values` using the `transform_iterator` as an index. 

This was done (as opposed to just transforming `values_view->begin<T>()`) because we also needed the corresponding `group_size`.